### DISCUSSION TITLE
How to emulate `git status` to see staged changes

### DISCUSSION THREAD

**Discussion by AndyBaay** [GitoxideLabs/gitoxide#1369] [General]
I've looked around and can't find an example of how to get the staged file changes. I'm looking for the equivalent of `git diff HEAD --name-only` or the "Changes to be committed" section of `git status`. For example:
```
$ git status
On branch main
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   README.txt
        new file:   new_file.sh
        new file:   subdir2/README.txt

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        ignored.txt
```
I want to get the 3 staged changes: `README.txt, new_file.sh, subdir2/README.txt`.

I've tried this:
```
            let index_file = repo.open_index()?;
            let state = gix::index::State::from(index_file);
            for entry in state.entries() {
                info!("{} -> {}", entry.path(&state), entry.stage());
            }
```
But that gives me all files committed and staged, with the same value `0` for `entry.stage()`.

Is there a way to get a list of the staged changes? Ideally I would be able to see the line-level diff, but file-level is fine too.

**Comment by Byron** [comment]
Here a diff between a tree (`HEAD^{tree}`) and the index would be required, or a diff between two indices, but none of these facilities currently exists.

I [have been meaning](https://github.com/Byron/gitoxide/pull/1368) to implement this for a quite a while now, but other things keep popping up.
